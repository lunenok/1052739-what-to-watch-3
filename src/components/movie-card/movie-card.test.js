import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const testMovie = {
  id: 1,
  title: `Bohemian Rhapsody`,
  genre: `Drama`,
  year: 2019,
  img: `bohemian-rhapsody.jpg`
};

it(`Should movie card render correctly`, () => {
  const tree = renderer
    .create(<MovieCard
      key={testMovie.id}
      filmInfo={testMovie}
      onMovieTitleClick={() => {}}
      onCardMouseHoverOn={() => {}}
      onCardMouseHoverOff={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
