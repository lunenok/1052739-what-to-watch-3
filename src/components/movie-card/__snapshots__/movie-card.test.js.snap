// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should movie card render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Bohemian Rhapsody"
      height={175}
      src="img/pulp-fiction.jpg"
      width={280}
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
      onClick={[Function]}
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
